spec/sql_overrides.rb

```diff
@@ -0,0 +1,60 @@
+require "rom"
+require "rom-sql"
+
+# Overrides to be done in rom-sql
+module ROM
+  module SQL
+    class Schema < ROM::Schema
+      class AttributesInferrer
+        def call(schema, gateway)
+          dataset = schema.name.dataset
+
+          columns = filter_columns(gateway.connection.schema(dataset))
+
+          inferred = columns.map do |(name, definition)|
+            type = type_builder.(definition)
+
+            attr_class.new(type.meta(source: schema.name), name: name) if type
+          end.compact
+
+          missing = columns.map(&:first) - inferred.map { |attr| attr.name }
+
+          [inferred, missing]
+        end
+      end
+    end
+
+    class Attribute < ROM::Attribute
+      def aliased(name)
+        super.meta(sql_expr: sql_expr.as(name))
+      end
+      alias_method :as, :aliased
+    end
+
+    class Function < ROM::Attribute
+      def name
+        self.alias || super
+      end
+    end
+
+    class ProjectionDSL < DSL
+      def method_missing(meth, *args, &block)
+        if schema.key?(meth)
+          schema[meth]
+        else
+          type = type(meth)
+
+          if type
+            if args.empty?
+              ::ROM::SQL::Function.new(type, name: :not_used, schema: schema)
```

Comment:
To be honest, not sure what is happening here. But seems that `name:` is not needed at all.

Reply:
I see. I guess that in the meantime, before facing that refactor, the only possibility is to have `:name` as optional in its `dry-initializer` definition.

Reply:
For now let's do this:

``` ruby
# FIXME: refactor so that it returns an intermediate function representation which
#              needs to a name
::ROM::SQL::Function.new(type, name: :__anonymous__, schema: schema)
```

I'll make sure this gets addressed before the release.

Reply:
@solnic I was now re-adapting `rom-sql` to new changes, and I was adding `name: :__anonymous` everywhere a function is created without name. The problem is that `AS "__anonymous__"` gets added to a lot of resulting SQL queries.

Maybe we should tackle that refactor first... or just stick with `nil` names in the meanwhile.

Reply:
Oh damn :/ My bad, apologies, I didn't think this through. Could you try redefining `option :name, optional: true, type: Types::Symbol` and see if that works?